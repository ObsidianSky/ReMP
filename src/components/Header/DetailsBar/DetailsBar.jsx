import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Bar from '../../common/Bar/Bar';

export const DetailsBar = ({ directorName }) => (
    <Bar title={directorName ? `Films by ${directorName}` : ''} />
);

DetailsBar.propTypes = {
    directorName: PropTypes.string
};

const mapStateToProps = ({ movies }) => ({
    // for some reasons we can have movie without director
    directorName: movies.selectedMovie.director && movies.selectedMovie.director.name,
});

export default connect(mapStateToProps)(DetailsBar);