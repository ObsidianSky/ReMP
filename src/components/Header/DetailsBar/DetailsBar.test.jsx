import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DetailsBar } from './DetailsBar';

describe('DetailsBar', () => {

    describe('when we do NOT have director name', () => {
        test('should renders correctly', () => {
            const wrapper = shallow(<DetailsBar />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when we have director name', () => {
        let props;

        beforeEach(() => {
            props = {
                directorName: 'directorName'
            }
        });

        test('should renders correctly', () => {
            const wrapper = shallow(<DetailsBar { ...props }/>);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});
