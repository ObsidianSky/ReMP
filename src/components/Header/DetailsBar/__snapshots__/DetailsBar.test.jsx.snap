// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsBar when we do NOT have director name should renders correctly 1`] = `
<Bar
  title=""
/>
`;

exports[`DetailsBar when we have director name should renders correctly 1`] = `
<Bar
  title="Films by directorName"
/>
`;
